warning: 3 uninitialized fields at the end of the constructor call

```                                                                                                                                                                                                                                                                                                                                                                                                                                        
/home/chiller/Desktop/git/ddnet/src/game/client/components/menus.h:710:46: warning: 3 uninitialized fields at the end of the constructor call [clang-analyzer-optin.cplusplus.UninitializedObject]                                                                                                                                                                                                                                                                                          
  710 |                         m_Slot(-1), m_Own(false) { m_aFilename[0] = 0; }                                                                                                                                                                                                                                                                                                                                                                                                            
      |                                                                   ^                                                                                                                                                                                                                                                                                                                                                                                                                 
/home/chiller/Desktop/git/ddnet/src/game/client/components/menus.h:703:8: note: uninitialized field 'this->m_Failed'                                                                                                                                                                                                                                                                                                                                                                        
  703 |                 bool m_Failed;                                                                                                                                                                                                                                                                                                                                                                                                                                                      
      |                      ^~~~~~~~                                                                                                                                                                                                                                                                                                                                                                                                                                                       
/home/chiller/Desktop/git/ddnet/src/game/client/components/menus.h:704:7: note: uninitialized field 'this->m_Time'                                                                                                                                                                                                                                                                                                                                                                          
  704 |                 int m_Time;                                                                                                                                                                                                                                                                                                                                                                                                                                                         
      |                     ^~~~~~                                                                                                                                                                                                                                                                                                                                                                                                                                                          
/home/chiller/Desktop/git/ddnet/src/game/client/components/menus.h:707:10: note: uninitialized field 'this->m_Date'                                                                                                                                                                                                                                                                         
  707 |                 time_t m_Date;                                                                                                                                                                                                                                                                                                                                                      
      |                        ^~~~~~                                                                                                                                                                                                                                                                                                                                                       
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:623:5: note: Assuming 'MsgType' is not equal to NETMSGTYPE_SV_KILLMSG                                                                                                                                                                                                                                                  
  623 |         if(MsgType == NETMSGTYPE_SV_KILLMSG)                                                                                                                                                                                                                                                                                                                                        
      |            ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                                                                                                                                                                         
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:623:2: note: Taking false branch                                                                                                                                                                                                                                                                                       
  623 |         if(MsgType == NETMSGTYPE_SV_KILLMSG)                                                                                                                                                                                                                                                                                                                                        
      |         ^                                                                                                                                                                                                                                                                                                                                                                           
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:634:10: note: Assuming 'MsgType' is not equal to NETMSGTYPE_SV_KILLMSGTEAM                                                                                                                                                                                                                                             
  634 |         else if(MsgType == NETMSGTYPE_SV_KILLMSGTEAM)                                                                                                                                                                                                                                                                                                                               
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                                                                                                                                                                
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:634:7: note: Taking false branch                                                                                                                                                                                                                                                                                       
  634 |         else if(MsgType == NETMSGTYPE_SV_KILLMSGTEAM)                                                                                                                                                                                                                                                                                                                               
      |              ^                                                                                                                                                                                                                                                                                                                                                                      
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:648:10: note: Assuming 'MsgType' is equal to NETMSGTYPE_SV_CHAT                                                                                                                                                                                                                                                        
  648 |         else if(MsgType == NETMSGTYPE_SV_CHAT)                                                                                                                                                                                                                                                                                                                                      
      |                 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                                                                                                                                                                       
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:648:7: note: Taking true branch                                                                                                                                                                                                                                                                                        
  648 |         else if(MsgType == NETMSGTYPE_SV_CHAT)                                                                                                                                                                                                                                                                                                                                      
      |              ^                                                                                                                                                                                                                                                                                                                                                                      
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:651:6: note: Assuming the condition is true                                                                                                                                                                                                                                                                            
  651 |                 if(pMsg->m_ClientId == -1 && m_Recording)                                                                                                                                                                                                                                                                                                                           
      |                    ^~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                                                                                                                                                                           
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:651:6: note: Left side of '&&' is true                                                                                                                                                                                                                                                                                 
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:651:32: note: Assuming field 'm_Recording' is true                                                                                                                                                                                                                                                                     
  651 |                 if(pMsg->m_ClientId == -1 && m_Recording)                                                                                                                                                                                                                                                                                                                           
      |                                              ^~~~~~~~~~~                                                                                                                                                                                                                                                                                                                            
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:651:3: note: Taking true branch                                                                                                                                                                                                                                                                                        
  651 |                 if(pMsg->m_ClientId == -1 && m_Recording)                                                                                                                                                                                                                                                                                                                           
      |                 ^                                                                                                                                                                                                                                                                                                                                                                   
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:655:7: note: Assuming 'Time' is > 0                                                                                                                                                                                                                                                                                    
  655 |                         if(Time > 0 && m_pClient->m_Snap.m_LocalClientId >= 0 && str_comp(aName, m_pClient->m_aClients[m_pClient->m_Snap.m_LocalClientId].m_aName) == 0)                                                                                                                                                                                                            
      |                            ^~~~~~~~                                                                                                                                                                                                                                                                                                                                                 
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:655:7: note: Left side of '&&' is true                                                                                                                                                                                                                                                                                 
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:655:19: note: Assuming field 'm_LocalClientId' is >= 0                                                                                                                                                                                                                                                                 
  655 |                         if(Time > 0 && m_pClient->m_Snap.m_LocalClientId >= 0 && str_comp(aName, m_pClient->m_aClients[m_pClient->m_Snap.m_LocalClientId].m_aName) == 0)                                                                                                                                                                                                            
      |                                        ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~                                                                                                                                                                                                                                                                                                       
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:655:7: note: Left side of '&&' is true                                                                                                                                                                                                                                                                                 
  655 |                         if(Time > 0 && m_pClient->m_Snap.m_LocalClientId >= 0 && str_comp(aName, m_pClient->m_aClients[m_pClient->m_Snap.m_LocalClientId].m_aName) == 0)                                                                                                                                                                                                            
      |                            ^                                                           
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:655:61: note: Assuming the condition is true                                                                              
  655 |                         if(Time > 0 && m_pClient->m_Snap.m_LocalClientId >= 0 && str_comp(aName, m_pClient->m_aClients[m_pClient->m_Snap.m_LocalClientId].m_aName) == 0)
      |                                                                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:655:4: note: Taking true branch                                                                                           
  655 |                         if(Time > 0 && m_pClient->m_Snap.m_LocalClientId >= 0 && str_comp(aName, m_pClient->m_aClients[m_pClient->m_Snap.m_LocalClientId].m_aName) == 0)
      |                         ^                                                              
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:657:5: note: Calling 'CGhost::StopRecord'                                                                                 
  657 |                                 StopRecord(Time);                                                              
      |                                 ^~~~~~~~~~~~~~~~                                                               
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:432:5: note: Assuming 'RecordingToFile' is false                                                                                                                          
  432 |         if(RecordingToFile)                        
      |            ^~~~~~~~~~~~~~~                         
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:432:2: note: Taking false branch                  
  432 |         if(RecordingToFile)                        
      |         ^                                          
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:436:5: note: 'Time' is > 0                        
  436 |         if(Time > 0 && (!pOwnGhost || Time < pOwnGhost->m_Time || !g_Config.m_ClRaceGhostSaveBest))                                                                                                                                    
      |            ^~~~                                    
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:436:5: note: Left side of '&&' is true                                                                                                                                    
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:436:18: note: Assuming 'pOwnGhost' is null                                                                                                                               
  436 |         if(Time > 0 && (!pOwnGhost || Time < pOwnGhost->m_Time || !g_Config.m_ClRaceGhostSaveBest))                                                                                                                                   
      |                         ^~~~~~~~~~                                                                             
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:436:29: note: Left side of '||' is true                                                                                                                                  
  436 |         if(Time > 0 && (!pOwnGhost || Time < pOwnGhost->m_Time || !g_Config.m_ClRaceGhostSaveBest))                                                                                                                                   
      |                                    ^                                                                           
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:440:6: note: Assuming the condition is false                                                                                                                             
  440 |                 if(Slot != -1 && (!pOwnGhost || Time < pOwnGhost->m_Time))                                     
      |                    ^~~~~~~~~~                                                                                  
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:440:17: note: Left side of '&&' is false                                                                                                                                 
  440 |                 if(Slot != -1 && (!pOwnGhost || Time < pOwnGhost->m_Time))                                     
      |                               ^                                                                                
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:443:6: note: 'pOwnGhost' is null                  
  443 |                 if(pOwnGhost && pOwnGhost->Active() && Time < pOwnGhost->m_Time)                               
      |                    ^~~~~~~~~                                                                                   
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:443:16: note: Left side of '&&' is false                                                                                                                                 
  443 |                 if(pOwnGhost && pOwnGhost->Active() && Time < pOwnGhost->m_Time)                               
      |                              ^                                                                                 
/home/chiller/Desktop/git/ddnet/src/game/client/components/ghost.cpp:447:22: note: Calling default constructor for 'CGhostItem'                                                                                                               
  447 |                 CMenus::CGhostItem Item;                                                                       
      |                                    ^~~~                                                                        
/home/chiller/Desktop/git/ddnet/src/game/client/components/menus.h:710:46: note: 3 uninitialized fields at the end of the constructor call                                                                                                    
  710 |                         m_Slot(-1), m_Own(false) { m_aFilename[0] = 0; }                                       
      |  
```

--- a/src/game/client/components/menus.h
+++ b/src/game/client/components/menus.h
@@ -700,11 +700,11 @@ public:
 		char m_aFilename[IO_MAX_PATH_LENGTH];
 		char m_aPlayer[MAX_NAME_LENGTH];
 
-		bool m_Failed = true;
-		int m_Time = 0;
-		int m_Slot = -1;
-		bool m_Own = false;
-		time_t m_Date = 0;
+		bool m_Failed;
+		int m_Time;
+		int m_Slot;
+		bool m_Own;
+		time_t m_Date;
 
 		CGhostItem() :
 			m_Slot(-1), m_Own(false) { m_aFilename[0] = 0; }

